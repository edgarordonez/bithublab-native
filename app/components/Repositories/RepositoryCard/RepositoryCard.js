// @flow

import React from 'react';
import { StyleSheet, View, TouchableOpacity, Image, Text } from 'react-native';
import type { Repository } from '@bithublab/types';
import ASSETS from '@bithublab/assets';

type Props = {
  repository: Repository,
  handleOnPress: (url?: string) => void,
};

export const RepositoryCard = ({ repository, handleOnPress }: Props) => {
  /* eslint-disable react/jsx-no-bind */
  const { provider, name, description, url } = repository;
  const onPressCard = () => handleOnPress(url);

  return (
    <TouchableOpacity onPress={onPressCard}>
      <View style={[styles.container, shadowStyle]}>
        <Image style={styles.image} source={ASSETS.images[provider]} />
        <View style={styles.info}>
          <Text style={styles.name}>{name}</Text>
          <Text style={styles.description}>{description}</Text>
        </View>
      </View>
    </TouchableOpacity>
  );
};

const shadowStyle = {
  shadowOpacity: 0.5,
  shadowRadius: 2,
  shadowColor: 'black',
  shadowOffset: { width: 1, height: 1 },
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    margin: 10,
    backgroundColor: '#fff',
    padding: 20,
    elevation: 2,
  },
  image: {
    width: 25,
    height: 25,
  },
  info: {
    paddingRight: 10,
    paddingLeft: 5,
  },
  name: {
    fontFamily: 'Roboto-Black',
    fontSize: 20,
    marginBottom: 5,
  },
  description: {
    fontFamily: 'Roboto-Regular',
    fontSize: 14,
  },
});
