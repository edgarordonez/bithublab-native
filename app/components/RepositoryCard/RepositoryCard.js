// @flow

import React from 'react';
import { StyleSheet, View, Image, Text } from 'react-native';

export const RepositoryCard = () => (
  <View style={[styles.container, shadowStyle]}>
    <Image
      style={styles.image}
      source={{
        uri:
          'https://assets-cdn.github.com/images/modules/logos_page/GitHub-Mark.png',
      }}
    />
    <View style={styles.info}>
      <Text style={styles.title}>edgarordonez/d3-stencil</Text>
      <Text style={styles.description}>
        Components of graphs build with StencilJS & D3.js
      </Text>
    </View>
  </View>
);

const shadowStyle = {
  shadowOpacity: 0.5,
  shadowRadius: 2,
  shadowColor: 'black',
  shadowOffset: { width: 1, height: 1 },
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    margin: 10,
    backgroundColor: '#fff',
    padding: 20,
  },
  image: {
    width: 25,
    height: 25,
  },
  info: {
    height: 80,
    paddingRight: 10,
    paddingLeft: 5,
  },
  title: {
    fontFamily: 'Roboto-Black',
    fontSize: 20,
    marginBottom: 5,
  },
  description: {
    fontFamily: 'Roboto-Regular',
    fontSize: 14,
  },
});
