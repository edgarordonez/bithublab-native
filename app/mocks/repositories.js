// @flow
import { Repository } from '@bithublab/components';

export const REPOSITORES: Repository[] = [
  {
    provider: 'github',
    name: 'D3-Stencil',
    description: 'Components of graphs build with StencilJS & D3.js',
    url: 'https://www.npmjs.com/package/d3-stencil',
  },
  {
    provider: 'gitlab',
    name: 'D3-Stencil',
    description: 'Components of graphs build with StencilJS & D3.js',
    url: 'https://www.npmjs.com/package/d3-stencil',
  },
  {
    provider: 'bitbucket',
    name: 'D3-Stencil',
    description: 'Components of graphs build with StencilJS & D3.js',
    url: 'https://www.npmjs.com/package/d3-stencil',
  }
];
