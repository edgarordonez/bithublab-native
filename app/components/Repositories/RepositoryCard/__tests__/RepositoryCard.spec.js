// @noflow

import React from 'react';
import TestRenderer from 'react-test-renderer';
import { RepositoryMock } from '@bithublab/mocks';

import { RepositoryCard } from '../RepositoryCard';

const Props = {
  repository: RepositoryMock,
  handleOnPress: jest.fn(),
};

describe('RepositoryCard', () => {
  it('Renders without crashing', () => {
    const rendered = TestRenderer.create(<RepositoryCard {...Props} />);
    expect(rendered).toMatchSnapshot();
  });
});
