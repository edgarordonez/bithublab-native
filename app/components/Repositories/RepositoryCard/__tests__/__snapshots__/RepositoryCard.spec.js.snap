// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryCard Renders without crashing 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "elevation": 2,
          "flexDirection": "row",
          "margin": 10,
          "padding": 20,
        },
        Object {
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 0.5,
          "shadowRadius": 2,
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 25,
          "width": 25,
        }
      }
    />
    <View
      style={
        Object {
          "paddingLeft": 5,
          "paddingRight": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontFamily": "Roboto-Black",
            "fontSize": 20,
            "marginBottom": 5,
          }
        }
      >
        edgarordonez/d3-stencil
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontFamily": "Roboto-Regular",
            "fontSize": 14,
          }
        }
      >
        Components of graphs build with StencilJS & D3.js
      </Text>
    </View>
  </View>
</View>
`;
